Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,388 @@
+            // Constants for creating the default starting location.
+            let networkSourceName = "Gas Device"
+            let assetGroupName = "Meter"
+            let assetTypeName = "Customer"
+            let terminalName = "Load"
+            let globalID = UUID(uuidString: "98A06E95-70BE-43E7-91B7-E34C9D3CB9FF")!
+            
+            // Create a default starting location.
+            if let networkSource = utilityNetwork.definition?.networkSource(named: networkSourceName),
+               let assetType = networkSource.assetGroup(named: assetGroupName)?.assetType(named: assetTypeName),
+               let startingLocation = utilityNetwork.makeElement(assetType: assetType, globalID: globalID) {
+                // Set the terminal for the location. (For our case, use the "Load" terminal.)
+                startingLocation.terminal = assetType.terminalConfiguration?.terminals.first(where: { $0.name == terminalName })
+                return startingLocation
+            } else {
+                return nil
+            }
+        }
+        
+        /// Adds a graphic to the graphics overlay.
+        /// - Parameters:
+        ///   - location: The `Point` location to place the graphic..
+        ///   - traceLocationType: The textual description of the trace location type.
+        private func addGraphic(for location: Point, traceLocationType: String) {
+            parametersOverlay.addGraphic(
+                Graphic(
+                    geometry: location,
+                    attributes: ["TraceLocationType": traceLocationType],
+                    symbol: nil
+                )
+            )
+        }
+        
+        /// Sets the selected filter barrier category and updates the status text.
+        func selectCategory(_ category: UtilityCategory) {
+            selectedCategory = category
+            traceEnabled = true
+            statusText = "\(category.name) selected."
+        }
+        
+        /// Runs a trace with the pending trace configuration and selects features in the map that
+        /// correspond to the element results.
+        @MainActor
+        func trace() async throws {
+            tracingActivity = .runningTrace
+            
+            guard let configuration = makeTraceConfiguration(category: selectedCategory) else {
+                statusText = "Failed to get trace configuration."
+                tracingActivity = .none
+                resetEnabled = true
+                return
+            }
+            traceParameters.traceConfiguration = configuration
+            var traceResults = [UtilityElementTraceResult]()
+            do {
+                traceResults = try await utilityNetwork
+                    .trace(using: traceParameters)
+                    .compactMap { $0 as? UtilityElementTraceResult }
+            } catch {
+                statusText = "Trace failed."
+                traceEnabled = true
+                traceCompleted = false
```

Comment:
for this to do what you think it's doing you'd want to return or rethrow. Returning would eat the error, which wouldn't be what you want I don't think.